tests/test_cloudpath_file_io.py

```diff
@@ -113,3 +117,26 @@ def test_os_open(rig):
     p = rig.create_cloud_path("dir_0/file0_0.txt")
     with open(p, "r") as f:
         assert f.readable()
+
+    with pytest.raises(BuiltInOpenWriteError):
+        with open(p, "w") as f:
+            pass
+
+    with pytest.raises(BuiltInOpenWriteError):
+        with open(p, "wb") as f:
+            pass
+
+    with pytest.raises(BuiltInOpenWriteError):
+        with open(p, "a") as f:
+            pass
+
+    with pytest.raises(BuiltInOpenWriteError):
+        with open(p, "r+") as f:
+            pass
+
+    # with pytest.raises(BuiltInOpenWriteError):
+    #     with open(
+    #         p,
+    #         "w",
+    #     ) as f:
+    #         pass
```

Comment:
This case fails because the test frame's `code_context` looks like this: `['        with open(\n']`. Not sure what to do about it.